File: java/src/org/futo/inputmethod/latin/uix/TextEditPopupActivity.java
package org.futo.inputmethod.latin.uix;

import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.os.Build;
import android.os.Bundle;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.EditText;
import android.widget.FrameLayout;

import androidx.annotation.ColorInt;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import org.futo.inputmethod.latin.R;

public class TextEditPopupActivity extends AppCompatActivity {

    @ColorInt
    private static int adjustAlpha(@ColorInt final int color, final float alpha) {
        final int a = Math.round(Color.alpha(color) * alpha);
        return (color & 0x00FFFFFF) | (a << 24);
    }

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        @ColorInt final int bgColor = Color.WHITE;
        @ColorInt final int fgColor = Color.BLACK;
        @ColorInt final int primaryColor = 0xFF6200EE;

        final EditText editText = new EditText(this);
        editText.setHint(R.string.settings_try_typing_here);
        editText.setBackgroundColor(bgColor);
        editText.setTextColor(fgColor);
        editText.setHintTextColor(adjustAlpha(fgColor, 0.7f));
        editText.setHighlightColor(adjustAlpha(primaryColor, 0.7f));
        editText.setMinLines(1);
        editText.setInputType(android.text.InputType.TYPE_CLASS_TEXT | android.text.InputType.TYPE_TEXT_FLAG_MULTI_LINE);

        if (Build.VERSION_CODES.Q <= Build.VERSION.SDK_INT) {
            Drawable d;
            if (null != (d = editText.getTextCursorDrawable())) d.setTint(primaryColor);
            if (null != (d = editText.getTextSelectHandle())) d.setTint(primaryColor);
            if (null != (d = editText.getTextSelectHandleLeft())) d.setTint(primaryColor);
            if (null != (d = editText.getTextSelectHandleRight())) d.setTint(primaryColor);
        }

        final CardView card = new CardView(this);
        card.setCardBackgroundColor(bgColor);
        card.setRadius(dp(16));
        card.setCardElevation(0);
        card.setUseCompatPadding(true);
        card.setContentPadding(dp(16), dp(16), dp(16), dp(16));
        card.addView(editText, new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));

        final FrameLayout root = new FrameLayout(this);
        root.setPadding(dp(8), dp(8), dp(8), dp(8));
        root.addView(card);

        setContentView(root);

        editText.requestFocus();
        getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
    }

    private int dp(final int dp) {
        return Math.round(dp * getResources().getDisplayMetrics().density);
    }
}

File: java/src/org/futo/inputmethod/latin/uix/TextEditPopupActivity.kt
package org.futo.inputmethod.latin.uix

import android.os.Bundle
import android.view.inputmethod.EditorInfo
import android.widget.EditText
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.toArgb
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.platform.LocalInspectionMode
import androidx.compose.ui.unit.dp
import androidx.compose.ui.viewinterop.AndroidView
import org.futo.inputmethod.latin.R

@Composable
fun AndroidTextInput() {
    val context = LocalContext.current
    val bgColor = MaterialTheme.colorScheme.background
    val fgColor = MaterialTheme.colorScheme.onBackground

    if(!LocalInspectionMode.current) {
        val editText = remember {
            EditText(context).apply {
                inputType = EditorInfo.TYPE_CLASS_TEXT
                isSingleLine = false
                this.

                setHint(R.string.try_typing)
                setBackgroundColor(bgColor.toArgb())
                setTextColor(fgColor.toArgb())
                setHintTextColor(fgColor.copy(alpha = 0.7f).toArgb())
            }
        }

        LaunchedEffect(bgColor, fgColor) {
            editText.setBackgroundColor(bgColor.toArgb())
            editText.setTextColor(fgColor.toArgb())
            editText.setHintTextColor(fgColor.copy(alpha = 0.7f).toArgb())
        }

        AndroidView({ editText }, modifier = Modifier
            .fillMaxWidth()
            .padding(16.dp), update = { editText.requestFocus() })
    }
}


class TextEditPopupActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContent {
            Surface(modifier = Modifier.padding(8.dp), shape = RoundedCornerShape(16.dp)) {
                AndroidTextInput()
            }
        }
    }

}
